src/apps/service/apps/de/job_view.clj

```diff
@@ -75,24 +99,42 @@
           (where {:app_id app-id})))
 
 (defn- format-steps
-  [app-id include-hidden-params?]
+  [user app-id include-hidden-params?]
   (let [app-steps         (get-steps app-id)
         multistep?        (> (count app-steps) 1)
         group-name-prefix (fn [{task-name :task_name}] (if multistep? (str task-name " - ") ""))]
-    (doall (mapcat (fn [step] (format-groups (group-name-prefix step) include-hidden-params? step)) app-steps))))
+    (doall (mapcat (fn [step] (format-groups user (group-name-prefix step) include-hidden-params? step)) app-steps))))
 
 (defn- format-app
-  [{app-id :id name :name :as app} include-hidden-params?]
+  [user {app-id :id name :name :as app} include-hidden-params?]
   (-> (select-keys app [:id :name :description :disabled :deleted])
       (assoc :label     name
-             :groups    (remove (comp empty? :parameters) (format-steps app-id include-hidden-params?))
+             :groups    (remove (comp empty? :parameters) (format-steps user app-id include-hidden-params?))
              :app_type  "DE"
              :system_id c/system-id)))
 
+(defn- validate-hidden-inputs [user app-id]
+  (when-let [paths (mapv :default_value (filter util/input? (mp/load-hidden-params app-id)))]
+    (try+
+     (data-info/get-path-info user :paths paths :validation-behavior "read")
```

Comment:
You should add a `:filter-include` here, probably `:filter-include "path"`. By only including that key, you minimize the amount of work that data-info has to do. With just `path` the only validation it'll do is that it exists and is readable, basically, and won't do anything like looking up UUIDs or info-types or etc. See also `apps.service.apps.jobs.util/create-output-dir` which does a similar `get-path-info` in a `try+` in order to check the existence/readability of an object in irods.

Reply:
Ah, good idea. Thanks for the suggestion.